cmd/monitor.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	ch := make(chan interface{})
+
+	switch mode := pathStat.Mode(); {
+	case mode.IsDir():
+		// Upload tree to remote
+		localPathClean := strings.TrimRight(localPath, "/") + "/"
+		remotePathClean := strings.TrimRight(remotePath, "/") + "/"
+
+		filesParams := make([]uploadParams, 0)
+		filepath.Walk(
+			localPathClean, func(path string, info os.FileInfo, err error) error {
+				if !info.Mode().IsRegular() {
+					return nil
+				}
+				relativePath := strings.SplitN(path, localPathClean, 2)[1]
+				remoteAbsoluteFilename := remotePathClean + relativePath
+				filesParams = append(filesParams, uploadParams{path, remoteAbsoluteFilename})
+				return nil
+			})
+
+		// Confirm user want to create those files in remote
+		fmt.Println("Following files are going to be created:")
+		for _, params := range filesParams {
+			fmt.Println(params.filePath + " -> " + params.remotePath)
+		}
+		var s string
+		fmt.Println("Confirm(y/n)?")
+		fmt.Scanln(&s)
+		if s != "y" {
+			return nil
+		}
+
+		go func() {
+			for _, params := range filesParams {
+				ch <- params
+			}
+			close(ch)
+		}()
+
+	case mode.IsRegular():
+		// Upload only one file to remote
+		go func() {
+			params := uploadParams{localPath, remotePath}
+			ch <- params
+			close(ch)
+		}()
+	default:
+		return errors.New("Not a regular file or folder")
+	}
+
+	client, err := NewAPIClient()
+	if err != nil {
+		return err
+	}
+
+	s := &monitorFileUpload{uploader: client.NewMonitorUploader()}
+	c := utils.NewCoordinator(viper.GetInt("threads"))
+	c.DoWithItemsFromChannel(s, ch)
+	return nil
+}
+
+var monitorItemUploadCmdHelp = `Upload a file or files contained in a folder.
+
+This command receives one file or folder path and uploads them to your
+VirusTotal Monitor account. It returns uploaded the file paths followed by their
+corresponding monitor ID.
+You can use the "vt monitor items [monitor_id]" command for retrieving
+information about the it.`
+
+var monitorItemUploadCmdExample = `  vt monitor item upload foo.exe /remote_folder/foo.exe
+  vt monitor item upload myfolder/ /another_remote_folder/`
+
+// NewMonitorItemsUploadCmd returns a new instance of the 'mointor upload file' command.
+func NewMonitorItemsUploadCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "upload [file/folder] [remote_path]",
+		Short:   "Upload one or more files to your account",
+		Long:    monitorItemUploadCmdHelp,
+		Example: monitorItemUploadCmdExample,
+		Args:    cobra.ExactArgs(2),
+		RunE:    runMonitorItemUpload,
+	}
+	addThreadsFlag(cmd.Flags())
+	return cmd
+}
+
+var monitorItemsCmdExample = `  vt monitor items list
+  vt monitor items list --filter "path:/myfolder/" --include path
+  vt monitor items list --filter "tag:detected" --include path,last_analysis_results.*.result,last_detections_count`
+
+// NewMonitorItemsCmd returns a new instance of the 'monitor_item' command.
+func NewMonitorItemsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "items [monitor_id] [flags]",
```

Comment:
You don't need to specify the `[flags]` part in the use string, this will be added automatically by cobra, `items [monitor_id]...` is enough. Notice that I'm adding ellipsis `...` to indicate that this command accepts multiple monitor IDs.

Reply:
done 92f8a89ddf28f4295f403dc802cb18254fecc136